float64 left
float64 right
float64 eLeft
float64 eRight
float64 cLeft
float64 cRight